model/infinispan/src/main/java/org/keycloak/jgroups/protocol/KEYCLOAK_JDBC_PING2.java

```diff
@@ -242,4 +242,58 @@ protected Connection getConnection() {
     public void setJpaConnectionProviderFactory(JpaConnectionProviderFactory factory) {
         this.factory = Objects.requireNonNull(factory);
     }
+
+    /**
+     * Detects a network partition and decides if the node belongs to the winning partition.
+     * <p>
+     * The algorithm performs the following steps
+     *
+     * <ul>
+     *     <li>Reads the data from the database</li>
+     *     <li>If an error occurs fetching the data, it returns {@link HealthStatus#ERROR}</li>
+     *     <li>Filters out non coordinator members</li>
+     *     <li>If no coordinator is found, it return {@link HealthStatus#NO_COORDINATOR}</li>
+     *     <li>If multiple coordinators are found, it compares them and uses the coordinator with the lowest {@link UUID}</li>
+     *     <li>Finally, it compares if the coordinator is the same as the current view coordinator. If so, it returns {@link HealthStatus#HEALTHY}, otherwise {@link HealthStatus#UNHEALTHY}</li>
+     * </ul>
+     *
+     * @return The {@link HealthStatus}.
+     * @see HealthStatus
+     */
+    public HealthStatus isHealthy() {
```

Comment:
As this doesn't return a boolean, I would rename to something like `healthStatus()` or `status()`.